frontend/projects/upgrade/src/app/features/dashboard/segments/pages/segment-details-page/segment-details-page.component.html

```diff
@@ -1,4 +1,14 @@
-<app-common-page>
-  <app-segment-details-page-header header></app-segment-details-page-header>
-  <app-segment-details-page-content content></app-segment-details-page-content>
-</app-common-page>
+<ng-container *ngIf="segment$ | async as segment">
+  <ng-container *ngIf="shouldUseLegacyView$ | async; else newView">
+    <!-- Legacy View -->
+    <view-segment></view-segment>
```

Comment:
i think this'll work for viewing, but what about the actions like adding / importing / editing segments that can be launched from the legacy view though? maybe we can re-wire those to the newer modals where possible when the toggle is on so we don't have to keep supporting those components?

Reply:
Only the details page will be displayed in the legacy view where you can't add or import a segment. I think we should use the legacy modal for editing legacy segments since they are very different (e.g., the legacy edit modal covers both overview details and segment members unlike the new design). 

I think there could be a possibility of displaying everything in the new segment details page though (so that we don't need to use the legacy view). Maybe we can discuss this in the meeting.